File: packages/react-native-video/android/src/main/java/androidx/media3/exoplayer/hls/HlsMediaSource.java
package androidx.media3.exoplayer.hls;

import androidx.media3.common.MediaItem;
import androidx.media3.datasource.DataSource;
import androidx.media3.exoplayer.drm.DrmSessionManagerProvider;
import androidx.media3.exoplayer.source.MediaSource;
import androidx.media3.exoplayer.upstream.LoadErrorHandlingPolicy;

public class HlsMediaSource {
    public static class Factory implements MediaSource.Factory {
        public Factory(DataSource.Factory mediaDataSourceFactory) {
        }

        @Override
        public MediaSource.Factory setDrmSessionManagerProvider(DrmSessionManagerProvider drmSessionManagerProvider) {
            return null;
        }

        @Override
        public MediaSource.Factory setLoadErrorHandlingPolicy(LoadErrorHandlingPolicy loadErrorHandlingPolicy) {
            return null;
        }

        @Override
        public int[] getSupportedTypes() {
            return new int[0];
        }

        @Override
        public MediaSource createMediaSource(MediaItem mediaItem) {
            return null;
        }

        public Factory setAllowChunklessPreparation(boolean allowChunklessPreparation) {
            return this;
        }
    }
}

File: packages/react-native-video/android/src/stubs/hls/androidx/media3/exoplayer/hls/HlsMediaSource.kt
package androidx.media3.exoplayer.hls

import androidx.media3.common.MediaItem
import androidx.media3.common.util.UnstableApi
import androidx.media3.datasource.DataSource
import androidx.media3.exoplayer.drm.DrmSessionManagerProvider
import androidx.media3.exoplayer.source.MediaSource
import androidx.media3.exoplayer.upstream.LoadErrorHandlingPolicy

class HlsMediaSource {
    @UnstableApi
    class Factory(private val mediaDataSourceFactory: DataSource.Factory) : MediaSource.Factory {
        override fun setDrmSessionManagerProvider(drmSessionManagerProvider: DrmSessionManagerProvider): MediaSource.Factory {
            throw UnsupportedOperationException("Not implemented")
        }

        override fun setLoadErrorHandlingPolicy(loadErrorHandlingPolicy: LoadErrorHandlingPolicy): MediaSource.Factory {
            throw UnsupportedOperationException("Not implemented")
        }

        override fun getSupportedTypes(): IntArray {
            return intArrayOf()
        }

        override fun createMediaSource(mediaItem: MediaItem): MediaSource {
            throw UnsupportedOperationException("Not implemented")
        }

        fun setAllowChunklessPreparation(allowChunklessPreparation: Boolean): Factory {
            return this
        }
    }
}
